homeassistant/components/media_player/__init__.py

```diff
@@ -78,38 +78,39 @@
 DEVICE_CLASSES_SCHEMA = vol.All(vol.Lower, vol.In(DEVICE_CLASSES))
 
 # Service call validation schemas
-MEDIA_PLAYER_SCHEMA = vol.Schema({
-    ATTR_ENTITY_ID: cv.comp_entity_ids,
-})
 
-MEDIA_PLAYER_SET_VOLUME_SCHEMA = MEDIA_PLAYER_SCHEMA.extend({
+# Make MEDIA_PLAYER_SCHEMA a reference to ENTITY_SERVICE_SCHEMA so that
+# platforms don't have to know about ENTITY_SERVICE_SCHEMA:
+MEDIA_PLAYER_SCHEMA = ENTITY_SERVICE_SCHEMA
```

Comment:
Should we update them to `ENTITY_SERVICE_SCHEMA` so we can get rid of this code smell?

Reply:
We certainly can – I was trying to avoid that, but if you think it's more appropriate, happy to accommodate.

Reply:
So platforms aren't supposed to be registering entry services anyway -- if they are using this new schema they have will have to manually implement the area implementation I believe?

Reply:
In some cases, they were latching onto and extending the base schemas used for the entity services. In most cases, those base schemas were replaced by `ENTITY_SERVICE_SCHEMA`. You bring up a good point: should user-defined services get to take advantage of a "base" laid out by the entity, or should they have to implement from scratch themselves?

Reply:
I think they should define their own schema.  The problem with extending is that when we add functionality (such as the `area_id`) they have to update their service calls to pick up and process the key.  This isn't an issue for our core services because entity service implements that, but platforms don't get to use that.

Reply:
Got it. 👍

Reply:
Using this as an example – https://github.com/home-assistant/home-assistant/blob/e4b4551b350a74b7b5a4dafd2f148a6387c14c01/homeassistant/components/monoprice/media_player.py#L92-L96 – would the desired result be to make those `register` calls us `ENTITY_SERVICE_SCHEMA`? Or are you saying that platforms shouldn't be touching that schema?

Reply:
Yeah, I agree with Andrew. It makes little sense to have platforms add support for AREA ID in their schema if they are not leveraging the entity component service processing.

Eventually ALL platform schemas are gone and are moved to integration specific. Like we did `media_player.sonos_x` becomes `sonos.x`

Reply:
Makes sense. Using the example of `monoprice` above (and a lot of other media players that do this), what should this PR do?

Reply:
If it’s not too much work, I’d just copy the original schema to the local platform.

Reply:
🙏